src/main/java/io/pravega/leaderelection/LeaderElection.java

```diff
@@ -0,0 +1,93 @@
+package io.pravega.leaderelection;
+import com.google.common.base.Preconditions;
+import io.pravega.client.ClientFactory;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.leaderelection.impl.LeaderElectionImpl;
+import lombok.extern.slf4j.Slf4j;
+import java.net.URI;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+@Slf4j
+public class LeaderElection {
+
+    private final LeaderElectionImpl leaderElectionImpl;
+    private final ClientFactory clientFactory;
+    private final StreamManager streamManager;
+    private final String scopeName;
+    private final String streamName;
+    private final String hostName;
+
+    public LeaderElection(String scope, String configName, URI controllerURI, String hostName, LeaderCrashListener listener) {
+        Preconditions.checkNotNull(scope);
+        Preconditions.checkNotNull(configName);
+        Preconditions.checkNotNull(controllerURI);
+        Preconditions.checkNotNull(listener);
+        Preconditions.checkNotNull(hostName);
+        this.scopeName = scope;
+        this.streamName = configName;
+        this.clientFactory = ClientFactory.withScope(scope, controllerURI);
+        this.streamManager = StreamManager.create(controllerURI);
+        this.hostName = hostName;
+        ScheduledExecutorService pool = Executors.newScheduledThreadPool(1);
+        leaderElectionImpl = new LeaderElectionImpl(scope, configName, clientFactory, streamManager, pool, listener, hostName);
+
+    }
+
+    public void start() {
+         leaderElectionImpl.add();
+         leaderElectionImpl.startAsync();
+
+    }
+
+    public void stop() {
+        leaderElectionImpl.stopAsync();
+    }
+
+    public Set<String> getCurrentMembers() {
+        return leaderElectionImpl.getCurrentMembers();
+    }
+
+    public String getCurrentLeader() {
+        return leaderElectionImpl.getCurrentLeader();
+    }
+
+    public String getInstanceId() {
+        return leaderElectionImpl.getInstanceId();
+    }
+
+    public Boolean isLeader() {
+        return hostName.equals(leaderElectionImpl.getCurrentLeader());
+    }
+
+    public interface LeaderCrashListener {
+        void selectLeader(String name);
+        void crashLeader(String name);
+    }
+
+    public void deleteStream() {
+        try {
+            streamManager.sealStream(scopeName, streamName);
+            Thread.sleep(500);
```

Comment:
Actually I am not sure whether the leader election should delete the stream. 
I think this should leave to the app, so I don't use this function right now.
When you create leaderelection object, if the scope and stream doesn't exist, it will create stream.
So I wonder if the delete scope and stream should let the leader election to do.